src/views/config-v2/AdditionalSettingsSection.tsx

```diff
@@ -52,6 +52,69 @@ export const AdditionalSettingsSection = (props: Props) => {
+    defaultTable: jsonData.defaultTable,
+    connMaxLifetime: jsonData.connMaxLifetime,
+    dialTimeout: jsonData.dialTimeout,
+    maxIdleConns: jsonData.maxIdleConns,
+    maxOpenConns: jsonData.maxOpenConns,
+    queryTimeout: jsonData.queryTimeout,
+    validateSql: jsonData.validateSql,
+    logs: {
+      defaultDatabase: jsonData.logs?.defaultDatabase,
+      defaultTable: jsonData.logs?.defaultTable !== 'otel_logs' ? jsonData.logs?.defaultTable : undefined,
+      otelEnabled: jsonData.logs?.otelEnabled,
+      otelVersion: jsonData.logs?.otelVersion !== 'latest' ? jsonData.logs?.otelVersion : undefined,
+      timeColumn: jsonData.logs?.timeColumn,
+      levelColumn: jsonData.logs?.levelColumn,
+      messageColumn: jsonData.logs?.messageColumn,
+      selectContextColumns: jsonData.logs?.selectContextColumns,
+      contextColumns:
+        jsonData.logs?.contextColumns && jsonData.logs?.contextColumns.length > 0
+          ? jsonData.logs?.contextColumns
+          : undefined,
+    },
+    traces: {
+      defaultDatabase: jsonData.traces?.defaultDatabase,
+      defaultTable: jsonData.traces?.defaultTable !== 'otel_traces' ? jsonData.logs?.defaultTable : undefined,
+      otelEnabled: jsonData.traces?.otelEnabled,
+      otelVersion: jsonData.traces?.otelVersion !== 'latest' ? jsonData.logs?.otelVersion : undefined,
+      traceIdColumn: jsonData.traces?.traceIdColumn,
+      spanIdColumn: jsonData.traces?.spanIdColumn,
+      operationNameColumn: jsonData.traces?.operationNameColumn,
+      parentSpanIdColumn: jsonData.traces?.parentSpanIdColumn,
+      serviceNameColumn: jsonData.traces?.serviceNameColumn,
+      durationColumn: jsonData.traces?.durationColumn,
+      startTimeColumn: jsonData.traces?.startTimeColumn,
+      tagsColumn: jsonData.traces?.tagsColumn,
+      serviceTagsColumn: jsonData.traces?.serviceTagsColumn,
+      kindColumn: jsonData.traces?.kindColumn,
+      statusCodeColumn: jsonData.traces?.statusCodeColumn,
+      statusMessageColumn: jsonData.traces?.statusMessageColumn,
+      stateColumn: jsonData.traces?.stateColumn,
+      instrumentationLibraryNameColumn: jsonData.traces?.instrumentationLibraryNameColumn,
+      instrumentationLibraryVersionColumn: jsonData.traces?.instrumentationLibraryVersionColumn,
+      flattenNested: jsonData.traces?.flattenNested,
+      traceEventsColumnPrefix: jsonData.traces?.traceEventsColumnPrefix,
+      traceLinksColumnPrefix: jsonData.traces?.traceLinksColumnPrefix,
+    },
+    aliasTables: jsonData.aliasTables && jsonData.aliasTables.length > 0 ? jsonData.aliasTables : undefined,
+    enableRowLimit: jsonData.enableRowLimit,
+    enableSecureSocksProxy: jsonData.enableSecureSocksProxy,
+  };
+
+  const hasDefinedValue = (obj: any): boolean => {
+    if (obj == null) {
+      return false;
+    }
+    if (typeof obj !== 'object') {
+      return obj !== '' && obj !== undefined && obj !== null;
+    }
+    return Object.values(obj).some((v) => hasDefinedValue(v));
```

Comment:
Nit: Is an object invalid if nested properties don't have a defined value? I think this function could do with a little documentation